### PR TITLE
Android support glb/gltf models

### PR DESCRIPTION
**Pull request by nickolanack**
added support for loading glb/~gltf~ models at runtime for android
I modified the example to use an online model
Screenshot_20190830-140950

this doesn't work correctly with gltf (but i think it is an issue with sceneform)

### CODE REVIEW

**Comment by nickolanack** [comment]
to clarify, .glb (like sfb) is the binary version of .gltf and includes embedded resources (materials and shape data) where .gltf is human readable version and may embed its resources as data urls, or reference external references relative to its location. 

.glb files can be placed in assets folder or be loaded from a url and seem to work fine. 

It appears that gltf files should behave the same way (according to the sceneform the api docs), however I have not had any success with it
- .gltf with embedded resources has a sceneformexception when reading data urls 
- .gltf with external resources has a null pointer exceptions `java.util.concurrent.CompletionException: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.ar.sceneform.assets.RenderableSource$SourceType.ordinal()' on a null object reference`

File: src/nodes/android/armodel.ts [hunk 1]
```diff
@@ -19,8 +32,25 @@ export class ARModel extends ARCommonGeometryNode {
...
     });
   }
+
+
+
+  static getGLTFSource(asset:string): any {
+    const context = utils.ad.getApplicationContext();
+
+
+        const type=(asset.indexOf(".glb")>0)?(<any>com.google.ar.sceneform).assets.RenderableSource.SourceType.GLB:(<any>com.google.ar.sceneform).assets.RenderableSource.SourceType.GLTF;
```
Comment (EddyVerbruggen) [inline src/nodes/android/armodel.ts]: @nickolanack According to https://developers.google.com/ar/reference/java/sceneform/reference/com/google/ar/sceneform/assets/RenderableSource.SourceType?hl=nl the `GLTF` enum is actually `GLTF2`. Would that perhaps fix the crash?

Comment (EddyVerbruggen) [reply src/nodes/android/armodel.ts]: On a related note: I'll try to generate TypeScript typings for those classes, so we can catch this kind of issues at code-time.

Comment (nickolanack) [reply src/nodes/android/armodel.ts]: Oh wow. Nice catch! I’ll give that a shot

Comment (nickolanack) [reply src/nodes/android/armodel.ts]: yup that fixes it!

**Comment by nickolanack** [comment]
Screenshot_20190831-101332
https://raw.githubusercontent.com/KhronosGroup/glTF-Sample-Models/master/2.0/SciFiHelmet/glTF/SciFiHelmet.gltf

**Comment by EddyVerbruggen** [comment]
That's one scary helmet! And super detailed as well. Wow.

**Comment by EddyVerbruggen** [comment]
The typings for the assets lib have been added, so once updated your IDE should have autocomplete for classes in the `com.google.ar.sceneform.assets` package.

**Pull request merged**